### PR TITLE
sse2,ssse3,sse4.1,avx512bw: added impl for mm512_s{l,r}l_epi16 + NEON impls

### PR DESCRIPTION
**Pull request by masterchef2209**
sse2,ssse3,sse4.1,avx512bw: added impl for mm512_s{l,r}l_epi16 + NEON impls

### CODE REVIEW

**Comment by masterchef2209** [comment]
Apart from drone failure of NEON impl. mm_set_epi64, apple osx/osx_xcode11 is failing native tests for mm512_srl_epi16, it is passing for all other architectures on travis. I am stuck on handling both these cases, please take a look.
Thank You

File: simde/x86/sse2.h [hunk 1]
```diff
@@ -4295,8 +4295,13 @@ simde_mm_set_epi64 (simde__m64 e1, simde__m64 e0) {
   #else
     simde__m128i_private r_;
 
-    r_.m64_private[0] = simde__m64_to_private(e0);
-    r_.m64_private[1] = simde__m64_to_private(e1);
+    #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
+      SIMDE_ALIGN_AS(16, int64x2_t) int64_t data[2] = {HEDLEY_STATIC_CAST(int64_t ,simde__m64_to_private(e0).neon_i64[0]), HEDLEY_STATIC_CAST(int64_t ,simde__m64_to_private(e1).neon_i64[0])};
```
[the remark this answers is not in the document]
Comment (masterchef2209) [reply simde/x86/sse2.h:4299]: thank you, I understand now

**Comment by masterchef2209** [comment]
all tests passed, I think this is fine now, I have added the 256-bit fallbacks and rebased changes on top of current master

File: simde/x86/avx512bw.h [hunk 2]
```diff
@@ -1930,6 +1930,46 @@ simde_mm512_sad_epu8 (simde__m512i a, simde__m512i b) {
...
+simde_mm512_sll_epi16 (simde__m512i a, simde__m128i count) {
+  #if defined(SIMDE_X86_AVX512BW_NATIVE)
+    return _mm512_sll_epi16(a, count);
+  #else
+    simde__m512i_private
+      r_,
+      a_ = simde__m512i_to_private(a);
+
+    #if SIMDE_NATURAL_VECTOR_SIZE_LE(256)
+        for (size_t i = 0 ; i < (sizeof(r_.m256i) / sizeof(r_.m256i[0])) ; i++) {
```
Comment (nemequ) [inline simde/x86/avx512bw.h:1944]: There is some extra indenting here.  It's not a big deal, and I'll fix it when I merge, but you might want to think about installing an [EditorConfig](https://editorconfig.org/) plugin so your editor picks up some configuration settings.  IIRC you use CLion, for which there is [a plugin](https://github.com/JetBrains/intellij-community/tree/master/plugins/editorconfig).

File: simde/x86/avx512bw.h [hunk 3]
```diff
@@ -1957,6 +1997,44 @@ simde_mm512_slli_epi16 (simde__m512i a, const int imm8)
...
+simde_mm512_srl_epi16 (simde__m512i a, simde__m128i count) {
+  #if defined(SIMDE_X86_AVX512BW_NATIVE)
+    return _mm512_srl_epi16(a, count);
+  #else
+    simde__m512i_private
+      r_,
+      a_ = simde__m512i_to_private(a);
+
+    #if SIMDE_NATURAL_VECTOR_SIZE_LE(256)
+        for (size_t i = 0 ; i < (sizeof(r_.m256i) / sizeof(r_.m256i[0])) ; i++) {
```
Comment (nemequ) [inline simde/x86/avx512bw.h:2011]: Same thing here

File: simde/x86/avx512bw.h (hunk 2 above)
Comment (masterchef2209) [reply simde/x86/avx512bw.h:1944]: I didn't know about the plugin for CLion, I have installed it now.
Thank You

**Pull request closed without merging**